### PR TITLE
ansible 2.9.9

### PR DESCRIPTION
**Pull request by dschanoeh**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
@dschanoeh thanks for the PR, can you check if we also need to update the resource dependencies?

**Comment by dschanoeh** [comment]
@chenrui333 sure, I can try. I haven't done this before but based on what I've read, it looks like I have to:

- create a venv and install everything under "Collect requirements from" using pip
- compute the resources using `poet -f ansible --also foo --also bar [...]`
- manually merge the updated resources into the formula and sort using poet_lint

Is that right or is there an easier way to do this?

**Comment by Moisan** [comment]
Another way to do it is to check on pypi.org if the resources are up to date one by one, whichever you think is faster.

**Comment by chenrui333** [comment]
The third way is to use this, `https://github.com/dawidd6/homebrew-update-pypi-resources`, this would help with the manual pypi manual check.

**Comment by dschanoeh** [comment]
All right, thanks for the help - the resources are also updated.

**Comment by Moisan** [comment]
Thank you for the resource update, could you squash your commits together with the original commit message (`ansible 2.9.9`)? Then we should be ready to merge.

**Comment by samford** [comment]
> The third way is to use this, `https://github.com/dawidd6/homebrew-update-pypi-resources`, this would help with the manual pypi manual check.

The last I checked, `homebrew-update-pypi-resources` simply used the latest version found on PyPi, which doesn't respect dependencies. This didn't work properly when I tested it on `ansible` in the past and I don't see any indication that anything has changed in the interim time to make it appropriate.

> Another way to do it is to check on pypi.org if the resources are up to date one by one, whichever you think is faster.

Updating the resources by hand (checking the latest version on PyPi) or using `poet` to identify the latest version of a package falls into the same trap as above.

The way to identify the correct versions with respect to what happens when you install `ansible` outside of Homebrew is like this: https://github.com/Homebrew/homebrew-core/pull/50504#issuecomment-592743222

It's a time-consuming process but it's sadly necessary to update the `ansible` formula properly.

File: Formula/ansible.rb [hunk 1]
```diff
@@ -138,13 +138,13 @@ class Ansible < Formula
   end
 
   resource "boto3" do
-    url "https://files.pythonhosted.org/packages/42/ec/7c7383defbaf6cb51b774219c63905b61e977983cab983959c873f6c0f91/boto3-1.12.46.tar.gz"
-    sha256 "4f7f11e7489c267f9ea0c6193bfbe20ef1a8dd639d780aea0ac61a769de722e1"
+    url "https://files.pythonhosted.org/packages/ee/b9/482d396fdbfab0e83f4c5f11b46a74b01a3ccdfdd84d54dd069083670b52/boto3-1.13.11.tar.gz"
+    sha256 "a7c5c7251b76336e697ccf368f125720e1947d58b218c228b61b5b654187fe4e"
```
Comment (samford) [inline Formula/ansible.rb:142]:
```suggestion
    url "https://files.pythonhosted.org/packages/e9/0a/7a512c839cf5c67f38571db41d2a68e12e577b346e7e87328490d4a2cbc3/boto3-1.13.12.tar.gz"
    sha256 "c46f31f085de660b95162c66057bae9ebb1658a245c9210162cee9671b0ff678"
```

This and `botocore` were just updated, so this is the correct version as of this moment.

File: Formula/ansible.rb [hunk 2]
```diff
@@ -138,13 +138,13 @@ class Ansible < Formula
...
   end
 
   resource "botocore" do
-    url "https://files.pythonhosted.org/packages/2b/d9/d1a341bec0f264eddba169b0d40aaa6da064fabba3e4a0d482b149c6fb84/botocore-1.15.46.tar.gz"
-    sha256 "ae39899dc450570968e3e128d2c8a011e5f8d4f6fd07a7d1f01df10a11ef332d"
+    url "https://files.pythonhosted.org/packages/ec/ce/12ad4123cc3dc0284408ad6dead2f0490ec38998a3b01a357e04106b655c/botocore-1.16.11.tar.gz"
+    sha256 "b82083f1ba65624017d53fa2d2a44aa801ea3da0948fba56ccf4d29a88ef0b71"
```
Comment (samford) [inline Formula/ansible.rb:147]:
```suggestion
    url "https://files.pythonhosted.org/packages/67/61/360e71652ce2596af042e7f2310373c653ec75a3e77dd11f5044bced9870/botocore-1.16.12.tar.gz"
    sha256 "b75a5dc97f9ac795139ea2c651c07a8522e31dc280db17243e2b20e5c210547b"
```

File: Formula/ansible.rb [hunk 3]
```diff
@@ -173,8 +173,8 @@ class Ansible < Formula
   end
 
   resource "cmd2" do
-    url "https://files.pythonhosted.org/packages/21/48/d48fe56f794e9a3feef440e4fb5c80dd4309575e13e132265fc160e82033/cmd2-0.8.9.tar.gz"
-    sha256 "145cb677ebd0e3cae546ab81c30f6c25e0b08ba0f1071df854d53707ea792633"
+    url "https://files.pythonhosted.org/packages/ba/b4/95e99237786d15572c9f80a279799964c5e910fc8c7d40c82dd5b4053108/cmd2-1.0.2.tar.gz"
+    sha256 "42ebf468bd457d99b8d2fd9573a8fba8519f625ca20d2d8185dbfbc416bd28b8"
```
Comment (samford) [inline Formula/ansible.rb:177]:
```suggestion
    url "https://files.pythonhosted.org/packages/21/48/d48fe56f794e9a3feef440e4fb5c80dd4309575e13e132265fc160e82033/cmd2-0.8.9.tar.gz"
    sha256 "145cb677ebd0e3cae546ab81c30f6c25e0b08ba0f1071df854d53707ea792633"
```

The correct version here is 0.8.9.

File: Formula/ansible.rb [hunk 4]
```diff
@@ -218,13 +218,13 @@ class Ansible < Formula
   end
 
   resource "docutils" do
-    url "https://files.pythonhosted.org/packages/93/22/953e071b589b0b1fee420ab06a0d15e5aa0c7470eb9966d60393ce58ad61/docutils-0.15.2.tar.gz"
-    sha256 "a2aeea129088da402665e92e0b25b04b073c04b2dce4ab65caaa38b7ce2e1a99"
+    url "https://files.pythonhosted.org/packages/2f/e0/3d435b34abd2d62e8206171892f174b180cd37b09d57b924ca5c2ef2219d/docutils-0.16.tar.gz"
+    sha256 "c2de3a60e9e7d07be26b7f2b00ca0309c207e06c100f9cc2a94931fc75a478fc"
```
Comment (samford) [inline Formula/ansible.rb:222]:
```suggestion
    url "https://files.pythonhosted.org/packages/93/22/953e071b589b0b1fee420ab06a0d15e5aa0c7470eb9966d60393ce58ad61/docutils-0.15.2.tar.gz"
    sha256 "a2aeea129088da402665e92e0b25b04b073c04b2dce4ab65caaa38b7ce2e1a99"
```

The correct version here is 0.15.2.

File: Formula/ansible.rb [hunk 5]
```diff
@@ -233,8 +233,8 @@ class Ansible < Formula
   end
 
   resource "google-auth" do
-    url "https://files.pythonhosted.org/packages/d8/47/0b6f9d832fe0699c8daf8b645408752e995f5afbb466cfa76d9954fcf8e1/google-auth-1.14.1.tar.gz"
-    sha256 "e63b2210e03c4ed829063b72c4af0c4b867c2788efb3210b6b9439b488bd3afd"
+    url "https://files.pythonhosted.org/packages/6c/39/333c46a06765856acca87f12ea93a48ae17eadb032ebe33f9f6db461bb01/google-auth-1.14.3.tar.gz"
+    sha256 "82d32f6601f35309c95d5917bcdf72f3ec193c7f46c79e433b8d76ccbe68e21d"
```
Comment (samford) [inline Formula/ansible.rb:237]:
```suggestion
    url "https://files.pythonhosted.org/packages/06/c1/54ebb8fe45524c0f84ff56a772e80b7a65975ff3825139ab604182da01be/google-auth-1.15.0.tar.gz"
    sha256 "fbf25fee328c0828ef293459d9c649ef84ee44c0b932bb999d19df0ead1b40cf"
```

This was updated in the interim time (a few hours ago), so the correct version is now 1.15.0.

**Review by samford — CHANGES_REQUESTED** [review]
Anything that I didn't specifically comment on here was correct. Thanks for putting in the work to update this!

**Comment by apple4ever** [comment]
@samford We did discuss `future` [in the last update](https://github.com/Homebrew/homebrew-core/pull/53661#pullrequestreview-400374055) . Its not needed for Python 3, and seemed to just be a remnant of how its checking. I also saw `terminal` the last time, but [looking it up](https://pypi.org/project/terminal/) didn't seem like it was needed

**Comment by samford** [comment]
> We did discuss `future` [in the last update](https://github.com/Homebrew/homebrew-core/pull/53661#pullrequestreview-400374055) . Its not needed for Python 3, and seemed to just be a remnant of how its checking. I also saw `terminal` the last time, but [looking it up](https://pypi.org/project/terminal/) didn't seem like it was needed

Ah, despite updating this formula sometimes, I don't actually use `ansible`, so I'll take your word that they aren't necessary.

**Comment by dschanoeh** [comment]
Thanks - didn't think the process was that complicated :). Do I squash these commits again or is it okay since that was a review?

**Comment by samford** [comment]
> Do I squash these commits again or is it okay since that was a review?

Please squash everything into one "ansible 2.9.9" commit and this will be good to go. Thanks!

**Review by samford — APPROVED** [review]

**Pull request closed without merging**